src/main/java/com/artipie/scripting/ScriptRunner.java

```diff
@@ -0,0 +1,101 @@
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobKey;
+import org.quartz.SchedulerException;
+import org.quartz.impl.StdSchedulerFactory;
+
+/**
+ * Script runner.
+ * Job for running script in quartz
+ * @since 0.30
+ */
+public final class ScriptRunner implements Job {
+
+    @Override
+    public void execute(final JobExecutionContext context) throws JobExecutionException {
+        final ScriptContext scontext = (ScriptContext) context
+            .getJobDetail().getJobDataMap().get("context");
+        final Key key = (Key) context.getJobDetail().getJobDataMap().get("key");
+        if (scontext == null || key == null) {
+            this.stopJob(context);
+            return;
+        }
+        if (scontext.getStorage().exists(key)) {
+            ScriptRunner.cachedScript(key, scontext)
+                .map(
+                    script -> {
+                        Optional<Script.Result> result;
+                        try {
+                            final Map<String, Object> vars = new HashMap<>();
+                            vars.put("_settings", scontext.getSettings());
+                            vars.put("_repositories", scontext.getRepositories());
+                            result = Optional.of(script.call(vars));
+                        } catch (final ScriptException exc) {
+                            Logger.error(
+                                ScriptRunner.class,
+                                "Execution error in script %s %[exception]s",
+                                key.toString(),
+                                exc
+                            );
+                            result = Optional.empty();
+                        }
+                        return result;
+                    }
+                );
+        } else {
+            Logger.warn(ScriptRunner.class, "Cannot find script %s", key.toString());
+        }
+    }
+
+    /**
+     * Stops the job and logs error.
+     * @param context Job context
+     */
+    private void stopJob(final JobExecutionContext context) {
+        final JobKey key = context.getJobDetail().getKey();
+        try {
+            Logger.error(this, String.format("Force stopping job %s...", key));
+            new StdSchedulerFactory().getScheduler().deleteJob(key);
+            Logger.error(this, String.format("Job %s stopped.", key));
+        } catch (final SchedulerException error) {
+            Logger.error(this, String.format("Error while stopping job %s", key));
+            throw new ArtipieException(error);
+        }
+    }
+
+    /**
+     * Returns cached script. Scripts objects are handled by LoadingCache.
+     * @param key Key in the `scontext` storage for the script.
+     * @param scontext Target scontext object.
+     * @return Returns cached script or empty Optional in case of error.
+     */
+    private static Optional<Script> cachedScript(final Key key, final ScriptContext scontext) {
```

Comment:
Caching looks unnecessary complex for me, could you please make it as simple as possible?

Reply:
We should have something like this
```
scontext.getScripts().get(key, lk -> {
            final String script = new String(scontext.getStorage().value(lk));
            return new Script.PrecompiledScript(script);
        });
```

Reply:
@dgarus , thanks, I agree - now that cache is local, we can rely on constant storage instance.